### PR TITLE
models: move modeled_types to its own crate

### PR DESCRIPTION
**Pull request by cbgbt**
**Description of changes:**

```
models: move modeled_types to its own crate

This allows settings extensions to be added to the Bottlerocket repository that use our existing modeled types without introducing a circular dependency between `models` and the extension.
```

**Testing done:**
* Smoke testing of an existing Bottlerocket variant
* No functional changes. If all CI builds succeed, this should be acceptable.

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

File: sources/models/modeled_types/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+[package]
+name = "modeled_types"
+version = "0.1.0"
+authors = []
```
Comment (zmrow) [inline sources/models/modeled_types/Cargo.toml]: No authors?

**Review by zmrow — APPROVED** [review]

File: sources/models/modeled_types/Cargo.toml (hunk 1 above)
Comment (cbgbt) [reply sources/models/modeled_types/Cargo.toml]: I think the models have been contributed to by the majority of Bottlerocket developers. I'm not sure I could fairly construct an authors list!

File: sources/models/Cargo.toml [hunk 2]
```diff
@@ -10,24 +10,12 @@ build = "build.rs"
...
 bottlerocket-release = { path = "../bottlerocket-release", version = "0.1" }
-indexmap = { version = "1", features = ["serde"] }
-lazy_static = "1"
 libc = "0.2"
 model-derive = { path = "model-derive", version = "0.1" }
-regex = "1"
-scalar = { path = "scalar", version = "0.1" }
-scalar-derive = { path = "scalar-derive", version = "0.1" }
-semver = "1"
+modeled_types = { path = "modeled_types", version = "0.1" }
```
Comment (bcressey) [inline sources/models/Cargo.toml]: For consistency? We tend to use the kebab form elsewhere.

```suggestion
modeled-types = { path = "modeled-types", version = "0.1" }
```

**Review by bcressey — APPROVED** [review]

**Comment by cbgbt** [comment]
^ Renames `modeled_types` to `modeled-types` per suggestion from @bcressey

**Review by etungsten — APPROVED** [review]

**Pull request merged**